lib/appoptics_apm/inst/rack.rb

```diff
@@ -40,10 +40,11 @@ def call(env)
         options = AppOpticsAPM::XTraceOptions.new(env['HTTP_X_TRACE_OPTIONS'], env['HTTP_X_TRACE_OPTIONS_SIGNATURE'])
         xtrace = AppOpticsAPM::XTrace.valid?(env['HTTP_X_TRACE']) ? (env['HTTP_X_TRACE']) : nil
         settings = AppOpticsAPM::TransactionSettings.new(url, xtrace, options)
+        profile_spans = AppOpticsAPM::Config['profiling'] == :enabled ? 1 : -1
```

Comment:
Can a profile have an issue and then isn't reported, ie. will the profile span always be sent if profiling is enabled?

For .NET the reporting of the profile span KV checks if a profile has been reported for the sampled request and reports either 0 or 1 if profiling is enabled. If profiling is disabled the .net agent also reports -1.

Reply:
To report 0 would require a lot of refactoring in the Ruby code and new communication between the c++ and Ruby code, which I may tackle when I get to the SDK
For now Lin allows to have only -1 or 1 reported, just so we can check if profiling is enabled or not